tempodb/encoding/iterator_multiblock.go

```diff
@@ -116,20 +116,25 @@ func (i *multiblockIterator) iterate(ctx context.Context) {
 			comparison := bytes.Compare(currentID, lowestID)
 
 			if comparison == 0 {
-				lowestObject, _, err = i.combiner.Combine(i.dataEncoding, currentObject, lowestObject)
-				if err != nil {
-					i.err.Store(fmt.Errorf("error combining while Nexting: %w", err))
-					return
-				}
-				b.clear()
+				lowestObjects = append(lowestObjects, currentObject)
+				lowestBookmarks = append(lowestBookmarks, b)
 			} else if len(lowestID) == 0 || comparison == -1 {
 				lowestID = currentID
-				lowestObject = currentObject
-				lowestBookmark = b
+				lowestObjects = [][]byte{currentObject}
+				lowestBookmarks = []*bookmark{b}
 			}
 		}
 
-		if len(lowestID) == 0 || len(lowestObject) == 0 || lowestBookmark == nil {
+		lowestObject, _, err := i.combiner.Combine(i.dataEncoding, lowestObjects...)
+		if err != nil {
+			i.err.Store(fmt.Errorf("error combining while Nexting: %w", err))
+			return
+		}
+		for _, b := range lowestBookmarks {
+			b.clear()
```

Comment:
I don't understand the logic here. Why are we appending to `lowestBookmarks` if we clear them before using them in the log? Could we use a counter instead?

Reply:
`lowestBookmarks` is a slice that contains bookmark structs that "point to" the next lowest object for compaction. Clearing them just tells them to move past their current object. The reason we do it here is b/c if the condition below hits and we `continue` then we need to be sure that we've moved these bookmarks to the next thing or we will just get in an infinite loop.

Honestly unsure about the quality of that log. Might make more sense to just log `len(lowestBookmarks)`. The only thing I really wanted was info about why the condition evaluated to true.

Reply:
The log message previously printed `lowestBookmark`, which mainly was to distinguish between nil or not (when not nil it just prints `unsupported object type`).  It would be similar here, printing nil/empty slice/nil/non-nil entries quite messily.... but quite frankly since we haven't determined root cause for the empty objects yet, I am ok with that and prefer it over a simple `len()`.